File: simcoe/src/main/java/com/prolificinteractive/simcoe/trackers/EventTracking.java
package com.prolificinteractive.simcoe.trackers;

import com.prolificinteractive.simcoe.TrackingResult;
import java.util.Map;

/**
 * The event tracking interface, used with analytics.
 */
public interface EventTracking extends AnalyticsTracking {

  /**
   * Tracks an event with a given provider.
   *
   * @param eventName The event name.
   * @param properties The associated properties.
   * @return The tracking result.
   */
  TrackingResult trackEvent(String eventName, Map<String, Object> properties);
}

File: simcoe/src/main/java/com/prolificinteractive/simcoe/trackers/EventTracking.kt
package com.prolificinteractive.simcoe.trackers

import com.prolificinteractive.simcoe.Properties
import com.prolificinteractive.simcoe.TrackingResult

/**
 * The event tracking interface, used with analytics.
 */
interface EventTracking : AnalyticsTracking {

  /**
   * Tracks an event with a given provider.
   *
   * @param eventName The event name.
   * @param properties The associated properties.
   * @return The tracking result.
   */
  fun trackEvent(eventName: String, properties: Properties?): TrackingResult
}
